DOC: Update ordering details and some translations

HaoZeke opened the pull request:
#### Brief description of what is fixed or changed

<!-- If this pull request fixes an issue, write "Fixes gh-NNNN" in that exact
format, e.g. "Fixes gh-1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open.

OR/AND

Describe your changes here
-->
~- Updated the order of names in the `en/about` page to be in alphabetical order (as promised by the page)~
- Clarified the ordering of names
- Updated the names and order of `{ja,pt}/about` pages

stefanv commented:
The old list was alphabetical by last name, which made sense. The new list is... not alphabetical :)

stefanv commented:
Last name ordering makes more sense to me; I recommend we leave as-is.

HaoZeke commented:
> Last name ordering makes more sense to me; I recommend we leave as-is.

Oops, updated the document for less confusion for me (and maybe others); also updated the translations.

stefanv reviewed (approved):
Alphabetizing last names is hard! There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.

Thanks for tidying up!

melissawm commented on content/ja/about.md:
```diff
@@ -12,26 +12,28 @@ NumPy は 、NumPyコミュニティやより広範な科学計算用Python コ
-- Allan Haldane
 - Charles Harris
 - Stephan Hoyer
+- Melissa Weber Mendonça
```
Well, there's the issue... Weber is a last name (my mother's side) so technically I should go after Stéfan I think? 

Otherwise, looks good :)

HaoZeke replied:
Excellent point, moved things around ^_^

stefanv commented:
Thanks!
